pkg/spoke/controllers/deletioncontroller/deletion_controller.go

```diff
@@ -112,41 +134,95 @@ func (m *StaleManifestDeletionController) syncManifestWork(ctx context.Context,
 		}
 	})
 
-	// no work to do if applied resources are not changed
 	if reflect.DeepEqual(manifestWork.Status.AppliedResources, appliedResources) {
+		// requeue the work if there exists deleting applied resources
+		if len(deletingResources) != 0 {
```

Comment:
this doesn't need to be inside this predicate, allowing an if/else above instead of the logic being 10 lines down from here.